mpisppy/opt/sc.py

```diff
@@ -58,16 +45,12 @@ def build_model_for_scenario(self,
 
         _assert_continuous(m)
 
-        active_obj = _get_active_objective(m)
+        active_obj = find_active_objective(m)
         active_obj.deactivate()
-        m._mpispypy_weighted_obj = pyo.Objective(expr=m._mpisppy_probability * active_obj.expr, sense=active_obj.sense)
-
-        nonant_vars = dict()
-        for node in m._mpisppy_node_list:
-            ndn = node.name
-            for ndx, v in enumerate(node.nonant_vardata_list):
-                nonant_vars[ndn, ndx] = v
+        m._mpisppy_model = pyo.Block()
```

Comment:
Final nit; I wasn't clear before. `SPBase.__init__` already attaches this block, you just have to use it.
https://github.com/Pyomo/mpi-sppy/blob/76132b5dc03fae54b2c305e7dff9e43d92b092aa/mpisppy/spbase.py#L477-L482

Reply:
Thanks again. Fixed!